bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/api/dto/clip2/helper/Setters.java

```diff
@@ -341,7 +348,43 @@ public static Resource setResource(Resource target, Resource source) {
                 targetTimedEffects.setDuration(duration);
             }
         }
-
         return target;
     }
+
+    /**
+     * Extract on/dimming/color fields from light and grouped light resources
+     * and merge them into separate resources.
+     */
+    public static Collection<Resource> mergeLightResources(Collection<Resource> resources) {
```

Comment:
`Setters.setResource` will not overwrite existing fields, so in principle, if two subsequent dimming values would arrive in sequence, the last one would not be taken. It doesn't matter much though since there are only three relevant fields to merge: on, dimming and color, so this logic can easily be handled without all the other merging done by `setResource`.